### PR TITLE
Validation for the Descriptive name page

### PR DESCRIPTION
**Pull request by lluli**
Next step of add table journey: [the validation for the Descriptive name page.](https://uktrade.atlassian.net/jira/software/projects/DT/boards/356?selectedIssue=DT-2056)

**How to Test** 
1. Head to /datasets
2. Add /add-table to the end of the dataset ID
3. Click through the flow to get to 'Give your table a descriptive name'
4. Try 'data', 'dataset', 'record', '_' and an empty string in the input field to confirm the error messages are displaying correctly. 
5. Enter a valid string to proceed, currently being redirected to '/' to be changed once the nest step in the flow is completed.

### CODE REVIEW

File: dataworkspace/dataworkspace/tests/datasets/add_table/test_views.py [hunk 1]
```diff
@@ -195,3 +195,118 @@ def test_classification_check_page(self):
...
+            published=True,
+            user_access_type=UserAccessType.REQUIRES_AUTHORIZATION,
+            information_asset_owner=self.user,
+            government_security_classification=2,
+        )
+        self.source = factories.SourceTableFactory.create(
+            dataset=self.dataset, schema="test", table="table1"
+        )
+
+    def test_descriptive_table_name_page(self):
```
Comment (ian-leggett) [inline dataworkspace/dataworkspace/tests/datasets/add_table/test_views.py:215]: I would just add a test for the page title and backlink.

File: dataworkspace/dataworkspace/tests/datasets/add_table/test_views.py [hunk 2]
```diff
@@ -195,3 +195,118 @@ def test_classification_check_page(self):
...
+        paragraph_text = soup.find("p").contents
+
+        assert response.status_code == 200
+        assert "Give your table a descriptive name" in header_one_text
+        assert (
+            "This should be a meaningful name that can help users understand what the table contains. It will show in the catalogue item under the 'Name' field in the data tables section."
+            in paragraph_text
+        )
+
+    def test_error_shows_when_descriptive_table_name_input_contains_data(self):#PASS
```
Comment (ian-leggett) [inline dataworkspace/dataworkspace/tests/datasets/add_table/test_views.py:234]: Could we remove the #pass comment

File: dataworkspace/dataworkspace/apps/datasets/add_table/forms.py [hunk 3]
```diff
@@ -17,3 +20,27 @@ def __init__(self, *args, **kwargs):
...
+class DescriptiveNameForm(GOVUKDesignSystemForm):
+
+    def clean_descriptive_name(self):
+        cleaned_data = super().clean()
+        descriptive_name = cleaned_data["descriptive_name"].lower()
+        if 'dataset' in descriptive_name:
+            raise ValidationError("Descriptive name cannot contain the word 'dataset'")
+        elif 'data' in descriptive_name:
+            raise ValidationError("Descriptive name cannot contain the word 'data'")
+        elif 'record' in descriptive_name:
```
Comment (CaitBarnard) [inline dataworkspace/dataworkspace/apps/datasets/add_table/forms.py:34]: Should this be plural? I'm just looking at the table name example as I'm building it now and it's plural there

Comment (lluli) [reply dataworkspace/dataworkspace/apps/datasets/add_table/forms.py:34]: I spoke to Ieuan an asked if it was just 'records' that should be flagged or if the 'record' should also be flagged and he said both, so 'record' will captures both cases.

**Review by CaitBarnard — APPROVED** [review]

**Review by ian-leggett — APPROVED** [review]

**Pull request merged**